### PR TITLE
feat: transfer polyx

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/Polymesh.ts [hunk 1]
```diff
@@ -84,6 +89,16 @@ export class Polymesh {
     }
   }
 
+  /**
+   * Transfer an amount of POLYX to a specified account
+   *
+   * @param args.to - account id that will receive the polyx
+   * @param args.amount - amount of polyx to be transferred
```
Comment (monitz87) [inline src/Polymesh.ts:96]: All caps for POLYX

File: src/api/procedures/transferPolyX.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'Insufficient balance to perform this action',
+    });
+  }
+
+  try {
+    const identityIds = await identity.keyToIdentityIds(stringToAccountKey(to, context));
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const _did = identityIds.unwrap().asUnique;
```
Comment (monitz87) [inline src/api/procedures/transferPolyX.ts:43]: You should check if the did has valid CDD

File: src/api/procedures/transferPolyX.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    this.addTransaction(tx.balances.transfer, {}, to, numberToBalance(val, context));
+  } catch (err) {
+    throw new PolymeshError({
+      code: ErrorCode.FatalError,
+      message: 'The destination account has not an associated identity',
```
Comment (monitz87) [inline src/api/procedures/transferPolyX.ts:49]: The destination account doesn't have an asssociated identity

Also, the code should be ValidationError

File: src/api/procedures/transferPolyX.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  if (val.isGreaterThan(freeBalance)) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'Insufficient balance to perform this action',
+    });
+  }
+
+  try {
+    const identityIds = await identity.keyToIdentityIds(stringToAccountKey(to, context));
```
Comment (monitz87) [inline src/api/procedures/transferPolyX.ts:41]: This call should be bundled up with the balance one in a Promise.all (remember to leave the comment for queryMulti)

File: src/api/procedures/transferPolyX.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  } = this;
+
+  const { to, amount: val } = args;
+
+  const freeBalance = await context.accountBalance();
+
+  if (val.isGreaterThan(freeBalance)) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'Insufficient balance to perform this action',
```
Comment (monitz87) [inline src/api/procedures/transferPolyX.ts:36]: Insufficient balance

File: src/api/procedures/__tests__/transferPolyX.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    return expect(
+      prepareTransferPolyX.call(proc, {
+        to: 'someAccount',
+        amount: new BigNumber(101),
+      })
+    ).rejects.toThrow('Insufficient balance to perform this action');
+  });
+
+  test('should throw an error if destination account has not an associated identity', () => {
```
Comment (monitz87) [inline src/api/procedures/__tests__/transferPolyX.ts:45]: doesn't have

File: src/api/procedures/__tests__/transferPolyX.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+    return expect(
+      prepareTransferPolyX.call(proc, { to: 'someAccount', amount: new BigNumber(99) })
+    ).rejects.toThrow('The destination account has not an associated identity');
+  });
+
+  test('should add a balance transfer transaction to the queue', async () => {
+    const to = 'someAccount';
+    const amount = new BigNumber(99);
+    const rawBalance = polkadotMockUtils.createMockBalance(amount.toNumber());
```
Comment (monitz87) [inline src/api/procedures/__tests__/transferPolyX.ts:59]: rawAmount

File: src/api/procedures/transferPolyX.ts [hunk 8]
```diff
@@ -28,27 +30,54 @@ export async function prepareTransferPolyX(
 
   const { to, amount: val } = args;
```
Comment (monitz87) [inline src/api/procedures/transferPolyX.ts:31]: why rename amount?

Comment (shuffledex) [reply src/api/procedures/transferPolyX.ts:31]: no reason

**Pull request merged**